timeout.c

```diff
@@ -172,12 +172,20 @@ void timeout_configure_IWDT(void) {
 	}
 }
 
-bool timeout_had_IWDG_reset(void) {
+/**
+ * Check if we've rebooted after a watchdog timer reset
+ *
+ * @param preserve_flag   set to true to allow subsequent check
```

Comment:
I like the school of thought on bools that they should answer a true/false question. So `should_preserve_flag` would be my preference.

Reply:
"should" doesn't instill much confidence imo, maybe "do" would be better, but I'm fine with whatever name people prefer